File: Study/BroadcastReceiver(네트워크)/MainActivity.java
package edu.ourincheon.broadcastreceiver;

import androidx.appcompat.app.AppCompatActivity;

import android.content.IntentFilter;
import android.net.wifi.WifiManager;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    public static TextView tv_state;
    private NetworkReceiver receiver;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tv_state = findViewById(R.id.tv_state);

        receiver = new NetworkReceiver();

        IntentFilter filter = new IntentFilter();
        filter.addAction(WifiManager.NETWORK_STATE_CHANGED_ACTION);
        registerReceiver(receiver,filter); //브로드캐스트리시버를 등록함.

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        //브로드 캐스트 리시버 해제!
        unregisterReceiver(receiver);

    }
}

File: Study(Kotlin)/Basic/MainActivity.kt
package edu.ourincheon.myapplication

import androidx.appcompat.app.AppCompatActivity //androidx는 확장된 extension 라이브러리임.
import android.os.Bundle //오래전부터 사용해온 안드로이드 라이브러리

class MainActivity : AppCompatActivity() { // 자식 클래스 : 부모클래스 , ()가 있다는 것은 생성자를 호출한 것 cf)인터페이스 상속
    override fun onCreate(savedInstanceState: Bundle?) { //fun은 함수를 의미. AppCompatActivity의 onCreate를 재정의(오버라이드) 하겠다는 것. 인자는 설명 생략
        super.onCreate(savedInstanceState) // super, 즉 AppCompatActivity에 있는 onCreate를 호출한 것임.
        setContentView(R.layout.activity_main) //화면에 보일 레이아웃 리소스 지정.
    }
}
